hibernate-core/src/main/java/org/hibernate/query/SelectionQuery.java

```diff
@@ -588,6 +590,24 @@ default Stream<R> stream() {
 	 */
 	SelectionQuery<R> setHibernateLockMode(LockMode lockMode);
 
+	/**
+	 * Apply a timeout to the corresponding database query.
+	 *
+	 * @param timeout The timeout to apply
+	 *
+	 * @return {@code this}, for method chaining
+	 */
+	SelectionQuery<R> setTimeout(Timeout timeout);
```

Comment:
Why here instead of on `CommonQueryContract`?

Reply:
`CommonQueryContract` is the base for things which don't support locking as well.

Reply:
Yes but they still support timeouts I believe, and it already declares `setTimeout(int)`.

Reply:
Oh, you're talking about just the timeout.  I thought you meant `setLockScope` too.  Yeah I'll move `setTimeout` there - its reasonable, like you sadi they already have the int form.

Reply:
Thanks.